haystack/core/component/component.py

```diff
@@ -240,6 +240,10 @@ def inner(method, sockets):
                 if param_name == "self" or param_info.kind in (Parameter.VAR_POSITIONAL, Parameter.VAR_KEYWORD):
                     continue
 
+                # We expect this one to have a different signature SyncStreamingCallbackT vs. AsyncStreamingCallbackT
+                if param_name == "streaming_callback":
+                    continue
+
```

Comment:
I have doubts about this change. Not about the change itself, but if we start to make exceptions on how we compare `run` and `run_async`, this could become problematic in the long run.

What is the main problem with always using `StreamingCallbackT`?

Reply:
I understand your point, which makes sense. At the same time, I'm not sure if we should change how we compare `run` and `run_async`.
(I may be too conservative, so feel free to involve other reviewers as well.)

Reply:
Hey @julian-risch and @mpangrazzi I'd appreciate an additional review on this topic. Happy to follow @anakin87's suggestion if we'd prefer to prioritize the comparison of the run and run_async signatures.

Reply:
I think @sjrl point makes sense. But skipping `streaming_callback` on run/run_async comparison would mean not actively validating it. What about e.g. making `_compare_run_methods_signatures` _aware_ of type diff for `streaming_callback`? I know that adding edge cases on comparison might not be the best option either, but if we properly keep track of it (adding required tests) it shouldn't be a big deal. WDYT? cc @anakin87 @julian-risch

Reply:
Makes sense